src/open_inwoner/accounts/views/login.py

```diff
@@ -0,0 +1,369 @@
+            self.reset_form = True
+            return self.get(request, *args, **kwargs)
+
+        return super().dispatch(request, *args, **kwargs)
+
+    def get_success_url(self):
+        redirect_to = self.request.POST.get(
+            self.redirect_field_name, self.request.GET.get(self.redirect_field_name, "")
+        )
+
+        # Ensure the user-originating redirection url is safe.
+        if not redirect_to or not is_safe_url(
+            url=redirect_to,
+            allowed_hosts=[
+                self.request.get_host(),
+            ],
+        ):
+            if hasattr(self, "user_cache") and self.user_cache.is_staff:
+                redirect_to = reverse("admin:index")
+            else:
+                redirect_to = settings.LOGIN_REDIRECT_URL
+
+        return redirect_to
+
+    def get(self, request, *args, **kwargs):
+        if not request.user.is_anonymous:
+            return HttpResponseRedirect(settings.LOGIN_REDIRECT_URL)
+
+        return super().get(request, *args, **kwargs)
+
+    def get_form_kwargs(self):
+        form_kwargs = super().get_form_kwargs()
+        form_kwargs.update(
+            {
+                "user": self.user_cache,
+                "request": self.request,
+            }
+        )
+        return form_kwargs
+
+    def form_valid(self, form):
+        self.log_in()
+        return super().form_valid(form)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        context.update(
+            {
+                "user_cache_key": self.request.GET.get("user"),
+                "message": self.message,
+                "sms_button_text": self.sms_button_text,
+            }
+        )
+
+        if getattr(self, "reset_form", None):
+            context["form"] = VerifyTokenForm(user=self.user_cache)
+
+        return context
+
+
+class ResendTokenView(ThrottleMixin, LogMixin, View):
+    throttle_visits = 25
+    throttle_period = 60 * 5
+    throttle_name = "new_token"
+    url_name = "verify_token"
+
+    @method_decorator(never_cache)
+    def dispatch(self, request, *args, **kwargs):
+        self.user_cache = None
+        UserModel = get_user_model()
+
+        if request.GET.get("user"):
+            try:
+                max_age = getattr(settings, "ACCOUNTS_USER_TOKEN_EXPIRE_TIME", 300)
+                self.user_cache = UserModel.objects.get(
+                    pk=signer.unsign(request.GET.get("user"), max_age=max_age)
+                )
+            except (UserModel.DoesNotExist, BadSignature, SignatureExpired):
+                pass
+        else:
+            self.user_cache = request.user
+        if not self.user_cache:
+            return HttpResponseRedirect(reverse("login"))
+
+        return super().dispatch(request, *args, **kwargs)
+
+    def get_verification_location(self, request, user):
+        url_part = reverse(self.url_name)
+        location = request.META.get("HTTP_REFERER", url_part)
```

Comment:
Thinking about the Gitbub scan result: do we actually need to look at HTTP_REFERER? Or can we just use the fallback it is doing here (with self.url_name)?